### PR TITLE
Align the `starting_time` of the processed fiber photometry data

### PR DESCRIPTION
**Pull request by weiglszonja**
After looking at the converted files I noticed that some recordings might be cropped (the picoscope data was longer than the processed data). After digging into their [processing code](https://github.com/DombeckLab/Azcorra2023/blob/main/Fiber%20photometry%20data%20analysis/Data%20pre%20processing/data_processing.m#L196) I found out that in some cases they were cropping the beginning of the recording and saving the crop point named "cropStart". 

I'm using this variable to align the starting time of the processed interface, so we are saving the processed behavior data and fiber photometry data with the correct starting time.

### CODE REVIEW

**Comment by weiglszonja** [comment]
Example [20200205-0001.nwb](https://us-east-2.console.aws.amazon.com/s3/object/catalystneuro-processing?region=us-east-2&bucketType=general&prefix=dombeck-lab-to-nwb/20200205-0001.nwb):
Screenshot 2024-05-23 at 15 37 04 (2)

**Comment by weiglszonja** [comment]
I'll first test this for all sessions and then reopen, sorry for pinging too early @alessandratrapani

File: src/dombeck_lab_to_nwb/azcorra2023/interfaces/azcorra2023_processedfiberphotometryinterface.py [hunk 1]
```diff
@@ -43,6 +43,20 @@ def __init__(
...
+        crop_point = processed_photometry_data["cropStart"]
+        # If the crop point is a list, it means that the end of the recording was cropped as well
+        self._crop_start = crop_point[0] if isinstance(crop_point, np.ndarray) else crop_point
+
+    def get_starting_time(self) -> float:
+        """
+        Return the starting time of the processed photometry data.
+        If the start of the picoscope recording had artefacts, the corrupted segment was cut of manually cropping point
+        was saved as "cropStart". We are using this value to align the starting time of the processed data with the
+        picoscope data.
```
Comment (weiglszonja) [inline src/dombeck_lab_to_nwb/azcorra2023/interfaces/azcorra2023_processedfiberphotometryinterface.py:55]:
```suggestion
        If the start of the picoscope recording had artefacts, the corrupted segment was cut off manually and the cropping point
        was saved as "cropStart". We are using this value to align the starting time of the processed data with the
        picoscope data.
```

**Comment by weiglszonja** [comment]
I think this is ready for review, also will add this as a note for the mid-way meeting to double check with them.

**Review by alessandratrapani — APPROVED** [review]
It is clear to me. Looks good!

File: src/dombeck_lab_to_nwb/azcorra2023/interfaces/azcorra2023_fiberphotometryinterface.py [hunk 2]
```diff
@@ -48,6 +51,24 @@ def __init__(
...
+
+    def get_timestamps(self, stub_test: bool = False) -> np.ndarray:
+        timestamps = self._timestamps if self._timestamps is not None else self.get_original_timestamps()
+        if stub_test:
+            return timestamps[:6000]
+        return timestamps
+
+    def set_aligned_timestamps(self, aligned_timestamps: np.ndarray) -> None:
+        self._timestamps = np.array(aligned_timestamps)
+
```
Comment (weiglszonja) [inline src/dombeck_lab_to_nwb/azcorra2023/interfaces/azcorra2023_fiberphotometryinterface.py:71]: @alessandratrapani FYI

**Pull request merged**